src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs

```diff
@@ -58,16 +58,86 @@ internal static async Task<GetSessionResult> GetSession(HttpClient httpClient, U
+                .WithBaseUri(apiUrl)
+                .WithHeader("X-Yoti-Auth-Id", sdkId)
+                .WithEndpoint(string.Format($"/v2/sessions/{0}", sessionId))
+                .WithQueryParam("appId", sdkId)
+                .WithHttpMethod(HttpMethod.Get)
+                .Build();
+
+            using (HttpResponseMessage response = await shareSessionlRequest.Execute(httpClient).ConfigureAwait(false))
+            {
+                if (!response.IsSuccessStatusCode)
+                {
+                    Response.CreateYotiExceptionFromStatusCode<DynamicShareException>(response);
+                }
+
+                var responseObject = await response.Content.ReadAsStringAsync();
+                var deserialized = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<GetSessionResult>(responseObject));
+
+                return deserialized;
+
+            }
+        }
+
+        internal static async Task<CreateQrResult> CreateQrCode(HttpClient httpClient, Uri apiUrl, string sdkId, AsymmetricCipherKeyPair keyPair, QrRequest qrRequestPayload)
+        {
+            Validation.NotNull(httpClient, nameof(httpClient));
+            Validation.NotNull(apiUrl, nameof(apiUrl));
+            Validation.NotNull(sdkId, nameof(sdkId));
+            Validation.NotNull(keyPair, nameof(keyPair));
+            Validation.NotNull(qrRequestPayload, nameof(qrRequestPayload));
+
+            string serializedQrCode = JsonConvert.SerializeObject(
+                qrRequestPayload,
+                new JsonSerializerSettings
+                {
+                    NullValueHandling = NullValueHandling.Ignore
+                });
+            byte[] body = Encoding.UTF8.GetBytes(serializedQrCode);
 
-            //byte[] body = Encoding.UTF8.GetBytes(serializedScenario);
+            Request qrRequest = new RequestBuilder()
+                .WithKeyPair(keyPair)
+                .WithBaseUri(apiUrl)
+                .WithHeader("X-Yoti-Auth-Id", sdkId)
+                .WithEndpoint($"/v2/sessions")
+                .WithQueryParam("appId", sdkId)
+                .WithHttpMethod(HttpMethod.Post)
+                .WithContent(body)
+                .Build();
+
+            using (HttpResponseMessage response = await qrRequest.Execute(httpClient).ConfigureAwait(false))
+            {
+                if (!response.IsSuccessStatusCode)
+                {
+                    Response.CreateYotiExceptionFromStatusCode<DynamicShareException>(response);
+                }
+
+                var responseObject = await response.Content.ReadAsStringAsync();
+                var deserialized = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<CreateQrResult>(responseObject));
+
+                return deserialized;
+
+            }
+        }
+
+        internal static async Task<GetSessionResult> GetQrCode(HttpClient httpClient, Uri apiUrl, string sdkId, AsymmetricCipherKeyPair keyPair, string qrCodeId)
```

Comment:
GetSessionResult is not the expected response for the GET endpoint. Please check the openAPI docs as this endpoint has 2 different SUCCESS responses (depending on the header Yoti-Share-Mobile-Token)
for web
```
{
  "id": "qr.v2.TCR1yNPVTECiGxyH95ix4g",
  "expiry": "2023-20-07T02:00:12Z",
  "policy": "CgoKBnNlbGZpZSgBEhkKFWNvdWxkLWJlLWFueXRoaW5nLXRiaCgBGAEgAQ==",
  "extensions": [],
  "session": {
    "id": "ss.v2.ChZyZUJzOXFTNFRlV1lSa3ZhMkJBN3F3EgdsZDUuZ2Jy",
    "status": "QR_CODE_SCANNED",
    "expiry": "2023-20-07T02:00:12Z"
  },
  "redirectUri": "/redirect"
}
```
for mobile
```
{
  "id": "qr.v2.TCR1yNPVTECiGxyH95ix4g",
  "expiry": "2023-20-07T02:00:12Z",
  "policy": "CgoKBnNlbGZpZSgBEhkKFWNvdWxkLWJlLWFueXRoaW5nLXRiaCgBGAEgAQ==",
  "extensions": [],
  "session": {
    "id": "ss.v2.ChZyZUJzOXFTNFRlV1lSa3ZhMkJBN3F3EgdsZDUuZ2Jy",
    "expiry": "2023-20-07T02:00:12Z"
  },
  "shareToken": "_jmBVDmqQuKuZVG9K9DRZw",
  "domain": "app.com",
  "app": {
    "id": "b9adcd92-bf90-44cf-9204-ca1a6d4c1ecd",
    "name": "AppName",
    "trusted": true
  },
  "org": {
    "name": "OrgName",
    "status": "VERIFIED"
  }
}
```

Reply:
Please correct the return type for `GetQrCode` task.